src/Pux/Router.purs

```diff
@@ -136,8 +138,10 @@ instance matchFunctor :: Functor Match where
     maybe Nothing (\t -> Just $ Tuple (fst t) (f (snd t))) $ r2t r
 
 instance matchAlt :: Alt Match where
-  alt (Match a) (Match b) = Match $ \r -> do
-    (a r) <|> (b r)
+  alt (Match a) (Match b) = Match $ \r ->
+    case a r of -- Manual implementation to avoid unnecessary evaluation of b r in case a r is true. PureScript is strict! God I love Haskell ;-)
+      Nothing -> b r
+      Just x  -> Just x
```

Comment:
This is nasty - if you rely on <|> for Maybe the arguments will be evaluated beforehand - therefore b r is called always -> This is pretty bad as b r contains all following choices!